### PR TITLE
Delete and Get FAQ endpoints added

### PR DESCRIPTION
**Pull request by naveen8801**
## Issue that this pull request solves

 Closes: #668 

## Proposed changes

* Added two new endpoints for API - /Getfaq and /deleteFaq
* /Getfaq will return all FAQs available in db.
* /deleteFaq will delete a particular FAQ from the db, it is basically a PUT request and its payload should be:
```
{
    userId
    faqId
}
```

## Types of changes

_Put an `x` in the boxes that apply_

- [x] New feature (non-breaking change which adds functionality)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

image
image

## Additional Info

This PR is made under SWOC Program

### CODE REVIEW

File: backend/app/routes/faq/DeleteFaq.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const to = require('await-to-js').default;
+const FAQ = require('../../models/faq');
```
Comment (Kajol-Kumari) [inline backend/app/routes/faq/DeleteFaq.js:2]: Please rename `FAQ` to `Faq`

File: backend/app/routes/faq/DeleteFaq.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const Admin = require('../../models/Admin');
+
+module.exports = async (req, res, next) => {
+  const userId = req.body.userId;
+  const faqId = req.body.faqId;
+  const [err, response] = await to(Admin.findById(userId));
+  if (err) {
+    const error = new ErrorHandler(constants.ERRORS.DATABASE, {
+      statusCode: 500,
+      message: 'No Admin',
```
Comment (Kajol-Kumari) [inline backend/app/routes/faq/DeleteFaq.js:14]: Please update the message with `You don't have the required permissions`

File: backend/app/routes/faq/DeleteFaq.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      errStack: err,
+    });
+    return next(error);
+  }
+  const [err2, response2] = await to(FAQ.findByIdAndDelete(faqId));
+  if (err2) {
+    const error = new ErrorHandler(constants.ERRORS.DATABASE, {
+      statusCode: 500,
+      message: 'No FAQ found',
```
Comment (Kajol-Kumari) [inline backend/app/routes/faq/DeleteFaq.js:23]: please replace the message as `Faq doesn't exist`

File: backend/app/routes/faq/GetFaq.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const FAQ = require('../../models/faq');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  const [err, response] = await to(FAQ.find());
+  if (err) {
+    const error = new ErrorHandler(constants.ERRORS.DATABASE, {
+      statusCode: 500,
+      message: 'Database Error',
```
Comment (Kajol-Kumari) [inline backend/app/routes/faq/GetFaq.js:11]: Please replace the message as `Mongo Error: Fetching Failed`

File: backend/app/routes/index.js [hunk 5]
```diff
@@ -5,11 +5,15 @@ const { emailTest } = require('./testRoutes/emailTest');
 const tinyURL = require('./tinyURL');
 const broadcast = require('./broadcast');
 const FAQ = require('./faq/post');
+const GetFaq = require('./faq/GetFaq');
```
Comment (Kajol-Kumari) [inline backend/app/routes/index.js:8]: please use camelCase in variable and usenames for eg thi sline can be replaced with - 
`const getFaq = require('./faq/get-faq');`

File: backend/app/routes/index.js [hunk 6]
```diff
@@ -5,11 +5,15 @@ const { emailTest } = require('./testRoutes/emailTest');
...
+const DeleteFaq = require('./faq/DeleteFaq');
```
Comment (Kajol-Kumari) [inline backend/app/routes/index.js:9]: Please replace above line with-
`const deleteFaq = require('./faq/delete-faq');`

and make the required change in `faq/index.js` file

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@naveen8801 Pease do the below changes-
1. Rename `DeleteFaq.js` to `deleteFaq.js`
2. Rename `GetFaq.js` to `deleteFaq.js`

Also, try not to include `package-lock.json` file or `yarn.lock` files while making the PR

**Comment by naveen8801** [comment]
Hey @Kajol-Kumari, made requested changes to filenames and variable names but can't able to ignore package-lock.json and yarn.lock, i would be thankful if you can help me out :)

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@naveen8801 Can you please mention somewhere in the PR description the program under which you have made this PR?

**Comment by naveen8801** [comment]
> @naveen8801 Can you please mention somewhere in the PR description the program under which you have made this PR?

Hi @Kajol-Kumari , added it in PR description, you can check, and made all the requested changes :)

**Review by Kajol-Kumari — APPROVED** [review]
@naveen8801 Thanks for making the requested changes.

LGTM 👍

**Pull request merged**